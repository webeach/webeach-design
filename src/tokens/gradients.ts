import {
  DarkModeColor,
  DarkModeColorKey,
  DarkModeColorType,
  LightModeColor,
  LightModeColorKey,
  LightModeColorType,
} from './colors';

export const DarkModeGradient = {
  primary1: [DarkModeColor.secondary1, DarkModeColor.primary1],
  primary2: [DarkModeColor.secondary2, DarkModeColor.primary2],
  primary3: [DarkModeColor.secondary3, DarkModeColor.primary3],
  primary4: [DarkModeColor.secondary4, DarkModeColor.primary4],
  primary5: [DarkModeColor.secondary5, DarkModeColor.primary5],
  primary6: [DarkModeColor.secondary6, DarkModeColor.primary6],
  primary7: [DarkModeColor.secondary7, DarkModeColor.primary7],
  primary8: [DarkModeColor.secondary8, DarkModeColor.primary8],
  primary9: [DarkModeColor.secondary9, DarkModeColor.primary9],
  primary10: [DarkModeColor.secondary10, DarkModeColor.primary10],
} satisfies Record<
  string,
  [DarkModeColorType[DarkModeColorKey], DarkModeColorType[DarkModeColorKey]]
>;

export const LightModeGradient = {
  primary1: [LightModeColor.secondary1, LightModeColor.primary1],
  primary2: [LightModeColor.secondary2, LightModeColor.primary2],
  primary3: [LightModeColor.secondary3, LightModeColor.primary3],
  primary4: [LightModeColor.secondary4, LightModeColor.primary4],
  primary5: [LightModeColor.secondary5, LightModeColor.primary5],
  primary6: [LightModeColor.secondary6, LightModeColor.primary6],
  primary7: [LightModeColor.secondary7, LightModeColor.primary7],
  primary8: [LightModeColor.secondary8, LightModeColor.primary8],
  primary9: [LightModeColor.secondary9, LightModeColor.primary9],
  primary10: [LightModeColor.secondary10, LightModeColor.primary10],
} satisfies Record<
  string,
  [LightModeColorType[LightModeColorKey], LightModeColorType[LightModeColorKey]]
>;

export type DarkModeGradientType = Readonly<typeof DarkModeGradient>;
export type DarkModeGradientKey = keyof DarkModeGradientType;

export type LightModeGradientType = Readonly<typeof LightModeGradient>;
export type LightModeGradientKey = keyof LightModeGradientType;

export type GradientKey = DarkModeGradientKey & LightModeGradientKey;
