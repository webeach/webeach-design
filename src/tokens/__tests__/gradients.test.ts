import { DarkModeGradient, LightModeGradient } from '../gradients';

import { gradientMapSchema } from './schemas/gradientMapSchema';

describe('Gradients validation', () => {
  test('Validate DarkModeGradient', () => {
    expect(() => gradientMapSchema.parse(DarkModeGradient)).not.toThrow();
  });

  test('Validate LightModeGradient', () => {
    expect(() => gradientMapSchema.parse(LightModeGradient)).not.toThrow();
  });

  test('DarkModeGradient and LightModeGradient should have identical keys', () => {
    const darkGradientKeys = Object.keys(DarkModeGradient).sort();
    const lightGradientKeys = Object.keys(LightModeGradient).sort();

    expect(darkGradientKeys).toEqual(lightGradientKeys);
  });

  test('Snapshot DarkModeGradient', () => {
    expect(DarkModeGradient).toMatchSnapshot();
  });

  test('Snapshot LightModeGradient', () => {
    expect(LightModeGradient).toMatchSnapshot();
  });
});
