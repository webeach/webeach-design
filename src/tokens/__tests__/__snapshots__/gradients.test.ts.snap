// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Gradients validation > Snapshot DarkModeGradient 1`] = `
{
  "primary1": [
    "hsl(200 71% 13%)",
    "hsl(300 45% 11%)",
  ],
  "primary10": [
    "hsl(196 86% 88%)",
    "hsl(299 77% 90%)",
  ],
  "primary2": [
    "hsl(198 70% 18%)",
    "hsl(300 49% 15%)",
  ],
  "primary3": [
    "hsl(197 69% 25%)",
    "hsl(300 48% 20%)",
  ],
  "primary4": [
    "hsl(201 70% 34%)",
    "hsl(300 49% 28%)",
  ],
  "primary5": [
    "hsl(201 70% 42%)",
    "hsl(300 47% 35%)",
  ],
  "primary6": [
    "hsl(202 71% 47%)",
    "hsl(300 43% 49%)",
  ],
  "primary7": [
    "hsl(200 85% 63%)",
    "hsl(302 59% 63%)",
  ],
  "primary8": [
    "hsl(199 86% 70%)",
    "hsl(302 67% 72%)",
  ],
  "primary9": [
    "hsl(198 85% 79%)",
    "hsl(301 72% 80%)",
  ],
}
`;

exports[`Gradients validation > Snapshot LightModeGradient 1`] = `
{
  "primary1": [
    "hsl(203 85% 95%)",
    "hsl(300 83% 95%)",
  ],
  "primary10": [
    "hsl(208 79% 29%)",
    "hsl(294 43% 22%)",
  ],
  "primary2": [
    "hsl(203 85% 89%)",
    "hsl(299 77% 90%)",
  ],
  "primary3": [
    "hsl(204 85% 84%)",
    "hsl(301 72% 80%)",
  ],
  "primary4": [
    "hsl(205 85% 79%)",
    "hsl(302 71% 71%)",
  ],
  "primary5": [
    "hsl(204 84% 73%)",
    "hsl(302 72% 62%)",
  ],
  "primary6": [
    "hsl(201 84% 55%)",
    "hsl(301 53% 55%)",
  ],
  "primary7": [
    "hsl(202 71% 49%)",
    "hsl(301 42% 49%)",
  ],
  "primary8": [
    "hsl(203 72% 42%)",
    "hsl(299 41% 40%)",
  ],
  "primary9": [
    "hsl(205 75% 36%)",
    "hsl(297 41% 31%)",
  ],
}
`;

exports[`Gradients validation Snapshot DarkModeGradient 1`] = `
{
  "primary1": [
    "hsl(200 71% 13%)",
    "hsl(300 45% 11%)",
  ],
  "primary10": [
    "hsl(196 86% 88%)",
    "hsl(299 77% 90%)",
  ],
  "primary2": [
    "hsl(198 70% 18%)",
    "hsl(300 49% 15%)",
  ],
  "primary3": [
    "hsl(197 69% 25%)",
    "hsl(300 48% 20%)",
  ],
  "primary4": [
    "hsl(201 70% 34%)",
    "hsl(300 49% 28%)",
  ],
  "primary5": [
    "hsl(201 70% 42%)",
    "hsl(300 47% 35%)",
  ],
  "primary6": [
    "hsl(202 71% 47%)",
    "hsl(300 43% 49%)",
  ],
  "primary7": [
    "hsl(200 85% 63%)",
    "hsl(302 59% 63%)",
  ],
  "primary8": [
    "hsl(199 86% 70%)",
    "hsl(302 67% 72%)",
  ],
  "primary9": [
    "hsl(198 85% 79%)",
    "hsl(301 72% 80%)",
  ],
}
`;

exports[`Gradients validation Snapshot LightModeGradient 1`] = `
{
  "primary1": [
    "hsl(203 85% 95%)",
    "hsl(300 83% 95%)",
  ],
  "primary10": [
    "hsl(208 79% 29%)",
    "hsl(294 43% 22%)",
  ],
  "primary2": [
    "hsl(203 85% 89%)",
    "hsl(299 77% 90%)",
  ],
  "primary3": [
    "hsl(204 85% 84%)",
    "hsl(301 72% 80%)",
  ],
  "primary4": [
    "hsl(205 85% 79%)",
    "hsl(302 71% 71%)",
  ],
  "primary5": [
    "hsl(204 84% 73%)",
    "hsl(302 72% 62%)",
  ],
  "primary6": [
    "hsl(201 84% 55%)",
    "hsl(301 53% 55%)",
  ],
  "primary7": [
    "hsl(202 71% 49%)",
    "hsl(301 42% 49%)",
  ],
  "primary8": [
    "hsl(203 72% 42%)",
    "hsl(299 41% 40%)",
  ],
  "primary9": [
    "hsl(205 75% 36%)",
    "hsl(297 41% 31%)",
  ],
}
`;
